src/main/java/sg/dex/starfish/impl/memory/MemoryJob.java

```diff
@@ -70,14 +74,23 @@ public boolean isCancelled() {
 	@Override
 	public Map<String,Object> get(long timeoutMillis, TimeUnit timeUnit) {
 		try {
+			this.setStatus(Status.succeeded);
```

Comment:
This looks wrong. Should set status to succeeded in the thread executing the job immediately after success.

Reply:
updated the code